app/forms/declaration/TransportInformation.scala

```diff
@@ -0,0 +1,162 @@
+      VehicleRegistrationNumber,
+      IATAFlightNumber,
+      AircraftRegistrationNumber,
+      EuropeanVesselIDNumber,
+      NameOfInlandWaterwayVessel
+    )
+
+  val mapping = Forms.mapping(
+    "inlandModeOfTransportCode" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.inlandTransportMode.error.incorrect",
+          isContainedIn(allowedModeOfTransportCodes)
+        )
+    ),
+    "borderModeOfTransportCode" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.borderTransportMode.error.incorrect",
+          isEmpty or isContainedIn(allowedModeOfTransportCodes)
+        )
+    ).verifying("supplementary.transportInfo.borderTransportMode.error.empty", _.isDefined)
+      .transform[String](
+        optValue => optValue.getOrElse(""),
+        borderModeOfTransportCode => Some(borderModeOfTransportCode)
+      ),
+    "meansOfTransportOnDepartureType" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.meansOfTransport.departure.error.incorrect",
+          isEmpty or isContainedIn(allowedMeansOfTransportTypeCodes)
+        )
+    ).verifying("supplementary.transportInfo.meansOfTransport.departure.error.empty", _.isDefined)
+      .transform[String](
+        optValue => optValue.getOrElse(""),
+        meansOfTransportOnDepartureType => Some(meansOfTransportOnDepartureType)
+      ),
+    "meansOfTransportOnDepartureIDNumber" -> optional(
+      text()
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.length", noLongerThan(27))
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.specialCharacters", isAlphanumeric)
+    ),
+    "meansOfTransportCrossingTheBorderType" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.meansOfTransport.crossingTheBorder.error.incorrect",
+          isEmpty or isContainedIn(allowedMeansOfTransportTypeCodes)
+        )
+    ).verifying("supplementary.transportInfo.meansOfTransport.crossingTheBorder.error.empty", _.isDefined)
+      .transform[String](
+        value => value.getOrElse(""),
+        meansOfTransportCrossingTheBorderType => Some(meansOfTransportCrossingTheBorderType)
+      ),
+    "meansOfTransportCrossingTheBorderIDNumber" -> optional(
+      text()
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.length", noLongerThan(35))
+        .verifying("supplementary.transportInfo.meansOfTransport.idNumber.error.specialCharacters", isAlphanumeric)
+    ),
+    "meansOfTransportCrossingTheBorderNationality" -> optional(
+      text()
+        .verifying(
+          "supplementary.transportInfo.meansOfTransport.crossingTheBorder.nationality.error.incorrect",
+          isContainedIn(allCountries.map(_.countryName))
+        )
+    ),
+    "container" -> boolean
+  )(TransportInformation.apply)(TransportInformation.unapply)
+
+  object ModeOfTransportCodes {
```

Comment:
This ModeOfTransportCodes too. In TransportCodes

Reply:
the codes are all sorted don't worry about it